controllers/utils/cluster.go

```diff
@@ -54,3 +61,44 @@ func GetStorageClusterFromCurrentNamespace(ctx context.Context, c client.Client,
 	// Return the single StorageCluster
 	return &storageClusterList.Items[0], nil
 }
+
+func FetchStorageCluster(ctx context.Context, c client.Client, name, namespace string) (*ocsv1.StorageCluster, error) {
+	var sc ocsv1.StorageCluster
+	err := c.Get(ctx, types.NamespacedName{
+		Name:      name,
+		Namespace: namespace,
+	}, &sc)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return nil, fmt.Errorf("could not find StorageCluster %v in namespace %v: %w", name, namespace, err)
+		}
+		return nil, fmt.Errorf("failed to retrieve StorageCluster %v in namespace %v: %w", name, namespace, err)
+	}
+
+	return &sc, nil
+}
```

Comment:
Do we really need this util? It creates a abstraction that is not required

Reply:
>We already have utils to fetch resources, so this is no different.

Yes, we do have util to fetch resource in MCO, but eventually I would like to start getting rid of them as they don't add a lot of extra value, the code is much simpler to read and contribute without so many utils